### PR TITLE
[feature] #306: Validate links post-build

### LINKED ISSUE #306: Validate anchor links
It would be good if our documentation fails to build in case if we have broken anchor links between or within pages.

It is currently not easy (or even possible) to implement it using existing Vitepress API, so I created an upstream issue:

- ~~https://github.com/vuejs/vitepress/issues/2176~~ https://github.com/vuejs/vitepress/issues/646

### PR DESCRIPTION
**Pull request by 0x009922**
Closes #306

### CODE REVIEW

**Review by 6r1d — APPROVED** [review]
LGTM

File: src/guide/advanced/running-iroha-on-bare-metal.md [hunk 1]
```diff
@@ -455,7 +455,7 @@ Iroha in the real world.
 
 4.  Determine the web socket that the other peers will use to connect to
     you. Make sure that the port is open and use that address
-    ([`P2P_ADDR`](/guide/configure/peer-configuration.md#p2p_addr)) in your
+    ([`P2P_ADDR`](/guide/configure/peer-configuration.md#p2p-addr)) in your
     `configs/peer/config.json`.
```
Comment (WRRicht3r) [inline src/guide/advanced/running-iroha-on-bare-metal.md:459]:
```suggestion
    `configs/peer/config.json` file.
```

File: src/guide/blockchain/data-model.md [hunk 2]
```diff
@@ -85,13 +85,13 @@ domain_looking_glass *-- account_rabbit : registered in
 
 ::: details Language-specific guidance on registering these objects
```
Comment (WRRicht3r) [inline src/guide/blockchain/data-model.md:86]:
```suggestion
::: details Language-specific guides to register these objects
```

File: src/guide/build.md [hunk 3]
```diff
@@ -15,11 +15,11 @@ You need
 [^1]:
     If you're having issues with installing Rust compatible with our code
```
Comment (WRRicht3r) [inline src/guide/build.md:16]:
```suggestion
    If you're having issues installing Rust compatible with our code
```

File: src/guide/configure/peer-configuration.md [hunk 4]
```diff
@@ -15,7 +15,7 @@ Some of the configuration options are required, while others are used for
 fine-tuning. When you create a new peer, you are required to provide the
 following:
 
-- [Public and private keys for the peer](#iroha-keys) (`PUBLIC_KEY` and
+- [Public and private keys for the peer](#public-and-private-keys) (`PUBLIC_KEY` and
```
Comment (WRRicht3r) [inline src/guide/configure/peer-configuration.md:18]:
```suggestion
- [The peer's Public and private keys](#public-and-private-keys) (`PUBLIC_KEY` and
```

File: src/guide/glossary.md [hunk 5]
```diff
@@ -166,7 +166,7 @@ either transactions or registered event listeners. More on ISI
 This set of [isi](#iroha-special-instructions-isi) contains logical
 instructions like `If`, I/O related like `Notify` and compositions like
 `Sequence`. They are mostly used by
```
Comment (WRRicht3r) [inline src/guide/glossary.md:168]:
```suggestion
`Sequence`. They are mostly used as
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

**Comment by 0x009922** [comment]
@WRRicht3r, this PR is focused on functional changes and related documentation fixes. While I appreciate your suggestions on doc strings, they are technically outside the scope of this PR. I have included them this time, but in the future, I would prefer to keep PRs more granular.

**Pull request merged**